enhancements/alert-on-operator-updates.md

```diff
@@ -0,0 +1,158 @@
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+
+PodDisruptionBudget has three fields:
+
+- A label selector `.spec.selector` to specify the set of pods to which it applies. This field is required.
+- `.spec.minAvailable` which specifies the number of pods that must still be available after the eviction, even in the absence of the evicted pod.
+- `.spec.maxUnavailable` which specifies the number of pods that can be unavailable after the eviction
+
+Note: `minAvailable` & `maxUnavailable` can be either an absolute number or a percentage.
+
+An PDB is considered invalid if any of these fields has incorrect values and gets applied on to a cluster.
+
+Below lists a number of potential approaches:
+
+### 1. Add a command line flag (Straw Man proposal)
+The flag parses the PDB yaml and validates the fields 
+ 
+### 2. Validate PDB same way as the yaml files under `/manifests` folder in validation library (Straw Man proposal)
+This is similar to how OLM's existing manifests are validated as part of the validation library
+ 
+### 3. [Needs more investigation] Using [CUE](https://cuelang.org/docs/integrations/yaml/#validate-yaml-files)
+- CUE is a configuration language that allows you to define a `<pdb-checks>.cue` template for the PDB yaml
+- CUE can be used as a go library to define the template and validate the PDBs before being applied on to the cluster   
+
+#### Pros
+-  
+
+#### Cons
+- Not a lot to offer in terms of flexibility as compared to OPA (discussed next)
+ 
+### 4. OPA Gatekeeper
+
+##### Glossary 
+- Open Policy Agent ([OPA](https://www.openpolicyagent.org/docs/latest/)) is a policy engine for Cloud Native environments hosted by CNCF.
+- [Gatekeeper](https://github.com/open-policy-agent/gatekeeper) was created to enable users to **_customize_** admission control
+  via configuration, not code and to bring awareness of the cluster’s state, not just the single object under evaluation 
+  at admission time. It enforces policies executed by the Open Policy Agent (OPA).
+  
+#### What is it?
+- We know that an [admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/) 
+can be thought of as a gatekeeper that intercept (authenticated) API requests and may change the request object or deny the request altogether.
+OPA Gatekeeper can be thought of as an added layer that allows users to define a custom policy (called `Constraint Template`) which 
+the PDB gets validated against before it is applied on to the cluster.
+
+
+#### How it works?
+- The Gatekeeper project uses a admission webhook to process the admission request whenever a resource in the 
+  cluster is created, updated, or deleted.
+- During the validation process, Gatekeeper acts as a bridge between the API server and OPA. The API server will enforce all policies executed by OPA
+
+#### Steps: 
+1. Install a recent release of [Gatekeeper](https://github.com/open-policy-agent/gatekeeper/blob/master/deploy/gatekeeper.yaml)
```

Comment:
So we're saying that we would install another operator that OLM would use as a dependency here?

Reply:
This was a huge red flag for me too.  I wonder if it would be safer to import its APIs. It's not impossible to use them, but to say we can't validate because our dependent operator is down does not sound good.